app/src/main/java/com/eventyay/organizer/data/faq/FaqRepositoryImpl.java

```diff
@@ -18,31 +19,27 @@
 
     private final FaqApi faqApi;
     private final Repository repository;
-    private final RateLimiter<String> rateLimiter = new RateLimiter<>(Duration.ofMinutes(10));
+    private Context context;
 
     @Inject
-    public FaqRepositoryImpl(FaqApi faqApi, Repository repository) {
+    public FaqRepositoryImpl(FaqApi faqApi, Repository repository, Context context) {
         this.faqApi = faqApi;
         this.repository = repository;
+        this.context = context;
     }
 
     @Override
     public Observable<Faq> getFaqs(long eventId, boolean reload) {
-        Observable<Faq> diskObservable = Observable.defer(() ->
-            repository.getItems(Faq.class, Faq_Table.event_id.eq(eventId))
-        );
-
-        Observable<Faq> networkObservable = Observable.defer(() ->
-            faqApi.getFaqs(eventId)
-                .doOnNext(faqs -> repository.syncSave(Faq.class, faqs, Faq::getId, Faq_Table.id).subscribe())
-                .flatMapIterable(faqs -> faqs));
+        if (!repository.isConnected()) {
+            return Observable.error(new Throwable(Constants.NO_NETWORK));
+        }
 
-        return repository.observableOf(Faq.class)
-            .reload(reload)
-            .withRateLimiterConfig("Faqs", rateLimiter)
-            .withDiskObservable(diskObservable)
-            .withNetworkObservable(networkObservable)
-            .build();
+        return faqApi.getFaqs(eventId)
+            .doOnNext(faqs -> OrgaRoomDatabase
+                .getInstance(context).faqDao().getAllFaqs(eventId))
+            .flatMapIterable(faqs -> faqs)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread());
```

Comment:
This changes the behaviour. It aleays loads FAQs from API

Reply:
Please suggest how it should be modified.

Reply:
Fetch from DB and use switchIfEmpty

Reply:
```
return OrgaRoomDatabase
            .getInstance(context).faqDao().getAllFaqs(eventId)
            .switchIfEmpty(faqApi.getFaqs(eventId))
            .flatMapIterable(faqs -> faqs)
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread());
```

Does this look correct to you?

Reply:
Why do we need flatmap iterable?

Reply:
And no. This won't insert the FAQs in the DB when loaded from API

Reply:
> Why do we need flatmap iterable?

`switchIfEmpty()` will return a `Observable<List<Faq>>` while the function needs to return `Observable<Faq>`

Reply:
Make this function return Observable<List<Faq>>

Reply:
> And no. This won't insert the FAQs in the DB when loaded from API

But this isn't even displaying the FAQs from database, even after removing flatMapIterable() and change the function return type.
I've checked that the FAQs are being fetched properly from the database, just that the ProgressBar remains there and the list does not show.

Reply:
OK will fetch tomorrow and see. Before that, please change to Observable<List<Faq>>. Because of DB layer observable, the stream will never end. So Observable<Faq> onComplete will never get called and it can never be shown in the UI

Reply:
> And no. This won't insert the FAQs in the DB when loaded from API

And this would again require the use of flatMapIterable, won't it? To convert Observable<List<Faq>> to Observable<Faq> for passing into the insert function.

Reply:
> OK will fetch tomorrow and see. Before that, please change to Observable<List>. Because of DB layer observable, the stream will never end. So Observable onComplete will never get called and it can never be shown in the UI

Done.

Reply:
No, don't convert to Observable<Faq>

Reply:
Okay, then we'll have to loop through the list and insert items into the DB?

Reply:
No. https://developer.android.com/reference/androidx/room/Insert.html

Reply:
Okay we can insert the list directly

Reply:
Updated.

Reply:
@iamareebjamal Did you check this?

Reply:
Is it still not working after the change?

Reply:
No

Reply:
Logs:

```
android.database.sqlite.SQLiteConstraintException: UNIQUE constraint failed: resources.resourceid (code 1555)
        at android.database.sqlite.SQLiteConnection.nativeExecuteForLastInsertedRowId(Native Method)
        at android.database.sqlite.SQLiteConnection.executeForLastInsertedRowId(SQLiteConnection.java:782)
        at android.database.sqlite.SQLiteSession.executeForLastInsertedRowId(SQLiteSession.java:788)
        at android.database.sqlite.SQLiteStatement.executeInsert(SQLiteStatement.java:86)
        at android.database.sqlite.SQLiteDatabase.insertWithOnConflict(SQLiteDatabase.java:1474)
        at android.database.sqlite.SQLiteDatabase.insert(SQLiteDatabase.java:1343)
```

Reply:
Well, it can't insert the FAQs. Add OnConflictStrategy in Insert annotation

Reply:
It's already there.